Update documentation of read_csv to explain that index_col can be a string containg a column name

kimsey0 opened the pull request:
Uses the suggestion from #22276 and closes #22276.

kimsey0 committed:
Update documentation of read_csv to explain that index_col can be a string containg a column name.

gfyoung commented on doc/source/user_guide/io.rst:
```diff
@@ -110,11 +110,12 @@ names : array-like, default ``None``
-  MultiIndex is used. If you have a malformed file with delimiters at the end of
-  each line, you might consider ``index_col=False`` to force pandas to *not* use
-  the first column as the index (row names).
+index_col : int/string or sequence of int/string or ``False``, default ``None``
```
"int, str, sequence of int / str, or False" would be a slightly better way of wording this IMO.

kimsey0 replied:
I agree, and have implemented the change.

jreback commented on doc/source/user_guide/io.rst:
```diff
@@ -110,11 +110,12 @@ names : array-like, default ``None``
+  Column(s) to use as the row labels of the ``DataFrame``, either given as string name or column index.
+  If a sequence of int/string is given, a MultiIndex is used.
+  Columns used for the index (row names) are dropped from the actual columns of the input dataframe.
+  They are accessible via ``.index``.
```
this is not relevant here.

kimsey0 replied:
Just the line about indexes being accessible via `.index`, or also the one about them not being accessible as regular columns?

kimsey0 replied:
@jreback?

TomAugspurger replied:
L166 and L117 can probably be removed.

jreback commented on doc/source/user_guide/io.rst:
```diff
@@ -110,11 +110,12 @@ names : array-like, default ``None``
-  the first column as the index (row names).
+index_col : int/string or sequence of int/string or ``False``, default ``None``
+  Column(s) to use as the row labels of the ``DataFrame``, either given as string name or column index.
+  If a sequence of int/string is given, a MultiIndex is used.
```
your edit is not making this more clear, can you re-word.

kimsey0 replied:
I have just implemented the wording by @smcinerney in #22276. Perhaps you could suggest a better one?

kimsey0 replied:
@jreback?

kimsey0 committed:
Improve wording of documentation for index_col parameter in read_csv.

kimsey0 committed:
Remove unnecessary documentation

Remove description of columns used as indexes not also being present as normal columns in documentation for read_csv.

jreback commented on doc/source/user_guide/io.rst:
```diff
@@ -110,11 +110,10 @@ names : array-like, default ``None``
-  each line, you might consider ``index_col=False`` to force pandas to *not* use
-  the first column as the index (row names).
+index_col : int, str, sequence of int / str, or False, default ``None``
+  Column(s) to use as the row labels of the ``DataFrame``, either given as string name or column index.
```
can you wrap lines at 79 characters

kimsey0 replied:
Can do.

jreback commented on doc/source/user_guide/io.rst:
```diff
@@ -110,11 +110,10 @@ names : array-like, default ``None``
+index_col : int, str, sequence of int / str, or False, default ``None``
+  Column(s) to use as the row labels of the ``DataFrame``, either given as string name or column index.
+  If a sequence of int / str is given, a MultiIndex is used.
+  (Note: ``index_col=False`` can be used to force pandas to *not* use the first column as the index, e.g. when you have a malformed file with delimiters at the end of each line.)
```
can you remove the parens. you *can* use a blank line here I think to emphasize it. (and the Note itself is ok)

kimsey0 replied:
Done.

kimsey0 committed:
Improve formatting of read_csv documentation.

TomAugspurger commented on pandas/io/parsers.py:
```diff
@@ -102,11 +102,14 @@
+  string name or column index. If a sequence of int / str is given, a
+  MultiIndex is used. Columns used for the index (row names) are dropped from
+  the actual columns of the input dataframe. They are accessible via
+  ``.index``. (Note: ``index_col=False`` can be used to force pandas to *not*
```
Can you format this similar to the user_guide version?

kimsey0 replied:
Like this? (I'm not sure why this is duplicated across code comments and documentation in the first place instead of being included from the code when compiling the documentation, but if it's to be fixed, it should probably be in another pull request.)

kimsey0 committed:
Improve formatting of read_csv documentation.

TomAugspurger commented:
whitespace issue:
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=9437

On Thu, Mar 14, 2019 at 8:10 AM Jacob Bundgaard <notifications@github.com>
wrote:

> *@kimsey0* commented on this pull request.
> ------------------------------
>
> In pandas/io/parsers.py
> <https://github.com/pandas-dev/pandas/pull/25502#discussion_r265559601>:
>
> > @@ -102,11 +102,14 @@
>      List of column names to use. If file contains no header row, then you
>      should explicitly pass ``header=None``. Duplicates in this list will cause
>      a ``UserWarning`` to be issued.
> -index_col : int, sequence or bool, optional
> -    Column to use as the row labels of the DataFrame. If a sequence is given, a
> -    MultiIndex is used. If you have a malformed file with delimiters at the end
> -    of each line, you might consider ``index_col=False`` to force pandas to
> -    not use the first column as the index (row names).
> +index_col : int, str, sequence of int / str, or False, default ``None``
> +  Column(s) to use as the row labels of the ``DataFrame``, either given as
> +  string name or column index. If a sequence of int / str is given, a
> +  MultiIndex is used. Columns used for the index (row names) are dropped from
> +  the actual columns of the input dataframe. They are accessible via
> +  ``.index``. (Note: ``index_col=False`` can be used to force pandas to *not*
>
> Like this? (I'm not sure why this is duplicated across code comments and
> documentation in the first place instead of being included from the code
> when compiling the documentation, but if it's to be fixed, it should
> probably be in another pull request.)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/25502#discussion_r265559601>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQHIhZ-TsBxvM8nS56Oyv6ewcTJWsvOks5vWkpFgaJpZM4bY980>
> .
>

kimsey0 committed:
Remove tailing whitespace.

kimsey0 commented:
Oops! I didn't notice that there's no spaces on the empty lines. It's fixed now.

jreback commented:
lgtm. unless anyone has comments.

TomAugspurger commented on pandas/io/parsers.py:
```diff
@@ -102,11 +102,16 @@
+  Column(s) to use as the row labels of the ``DataFrame``, either given as
+  string name or column index. If a sequence of int / str is given, a
+  MultiIndex is used. Columns used for the index (row names) are dropped from
+  the actual columns of the input dataframe. They are accessible via
```
Hrm, I thought we decided to remove the "They are accessible via .index" bit.

kimsey0 replied:
I understood that it was only to be removed in one of the two places, but I have removed it in both now.

kimsey0 committed:
Remove unnecessary documentation

Remove description of columns used as indexes not also being present as normal columns in documentation for read_csv.

TomAugspurger commented:
Thanks @kimsey0!

kimsey0 commented:
Thanks for getting it merged.
